fcomm/tests/proof_tests.rs

```diff
@@ -90,28 +91,38 @@ fn test_open_commitment<T: AsRef<OsStr>>(
     cmd.assert().success();
 }
 
-fn test_verify_expression_proof<T: AsRef<OsStr>>(mut cmd: Command, proof_path: T, _data_path: T) {
+fn test_verify_expression_proof(
+    mut cmd: Command,
+    proof_path: &Utf8PathBuf,
+    _data_path: &Utf8PathBuf,
+) {
     cmd.arg("verify").arg("--proof").arg(proof_path);
 
     cmd.assert().success().stdout("{\"verified\":true}");
 }
 
-fn test_verify_opening<T: AsRef<OsStr>>(mut cmd: Command, proof_path: T, _data_path: T) {
+fn test_verify_opening(mut cmd: Command, proof_path: &Utf8PathBuf, _data_path: &Utf8PathBuf) {
     cmd.arg("verify").arg("--proof").arg(proof_path);
 
     cmd.assert().success().stdout("{\"verified\":true}");
 }
 
+#[inline]
+fn path_buf_to_utf8_path_buf(path_buf: std::path::PathBuf) -> Utf8PathBuf {
+    Utf8PathBuf::from_path_buf(path_buf.to_path_buf()).unwrap()
+}
```

Comment:
You could use this, or the `impl TryFrom<PathBuf> for Utf8PathBuf` but in any case:
- why convert a path to a pathbuf, do join operations on it, and then convert to a Utf8PathBuf, when the reason for `Utf8PathBuf` is to be a `String` with `PathBuf` semantics?
- why hide the unwrap behind an auxiliary function?

Consider:
```
       let tmp_dir = Utf8PathBuf::try_from(TempDir::new().unwrap()).unwrap();
       let foo_path = tmp_dir.join("...");
       let bar_path = ...
```